crates/astria-sequencer-client/src/extension_trait.rs

```diff
@@ -0,0 +1,458 @@
+//! Extension trait for making tendermint JSONRPCs specific to astria-sequencer.
+//!
+//! [`SequencerClientExt`] is implemented for [`tendermint_rpc::HttpClient`] and
+//! [`tendermint_rpc::WebSocketClient`], which are gated behind the features
+//! `http` and `websocket`, respectively.
+//!
+//! # Examples
+//! The example below works with the feature `"http"` set.
+//! ```no_run
+//! # tokio_test::block_on(async {
+//! use astria_sequencer_client::SequencerClientExt as _;
+//! use tendermint_rpc::HttpClient;
+//!
+//! let client = HttpClient::new("http://127.0.0.1:26657")?;
+//! let address: [u8; 20] = hex_literal::hex!("DEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF");
+//! let height = 5u32;
+//! let balance = client.get_balance(address, height).await?;
+//! println!("{balance:?}");
+//! # Ok::<_, Box<dyn std::error::Error>>(())
+//! # });
+//! ```
+
+use std::{
+    future,
+    pin::Pin,
+};
+
+use async_trait::async_trait;
+use futures::Stream;
+pub use proto::native::sequencer::v1alpha1::{
+    Address,
+    BalanceResponse,
+    NonceResponse,
+    SignedTransaction,
+};
+use sequencer_types::SequencerBlockData;
+#[cfg(feature = "http")]
+use tendermint_rpc::HttpClient;
+#[cfg(feature = "websocket")]
+use tendermint_rpc::WebSocketClient;
+use tendermint_rpc::{
+    endpoint::broadcast::{
+        tx_commit,
+        tx_sync,
+    },
+    event::EventData,
+    Client,
+    SubscriptionClient,
+};
+
+#[cfg(feature = "http")]
+impl SequencerClientExt for HttpClient {}
+#[cfg(feature = "websocket")]
+const _: () = {
```

Comment:
It allows me to group all these impls in one block. `{}` is not available at the module level, so this is not allowed:
```rust
# not allowed
#[cfg(feature = "websocket")]
{
    impl Foo for Bar {}
    impl Foo for Baz {}
}
```
This would work, but it's very repetitive:
```rust
# so much noise
#[cfg(feature = "websocket")]
impl Foo for Bar {}
#[cfg(feature = "websocket")]
impl Foo for Baz {}
```
Or this one, but there is a module in the namespace which has no right being there.
```rust
#[cfg(feature = "websocket")]
mod __impls {
    impl Foo for Bar {}
    impl Foo for Baz {}
}